src/bitcoind.cpp

```diff
@@ -105,24 +187,34 @@ static bool AppInit(int argc, char* argv[])
             // InitError will have been called with detailed error, which ends up on console
             return false;
         }
-        if (args.GetBoolArg("-daemon", false)) {
-#if HAVE_DECL_DAEMON
-#if defined(MAC_OSX)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#endif
+        if (args.GetBoolArg("-daemon", DEFAULT_DAEMON) || args.GetBoolArg("-daemonwait", DEFAULT_DAEMONWAIT)) {
+#if HAVE_DECL_FORK
             tfm::format(std::cout, PACKAGE_NAME " starting\n");
 
             // Daemonize
-            if (daemon(1, 0)) { // don't chdir (1), do close FDs (0)
-                return InitError(Untranslated(strprintf("daemon() failed: %s\n", strerror(errno))));
+            switch (fork_daemon(1, 0, daemon_ep)) { // don't chdir (1), do close FDs (0)
+                case 0: // Child: continue.
+                    // If -daemonwait is not enabled, immediately send a success token the parent.
+                    if (!args.GetBoolArg("-daemonwait", DEFAULT_DAEMONWAIT)) {
+                        daemon_ep.TokenWrite(true);
```

Comment:
nit, feel free to ignore: maybe avoid implicit casting from bool to uint8_t and use literal 1 (or some character literal) as success token instead? I think as it is this could be potentially confusing for readers (also the counterpart `TokenRead` call below). As a drawback though, `-daemonwait` code gets longer: `daemon_ep.TokenWrite(fRet ? 1 : 0)`.

Reply:
Agree, it's probably better to pass some constant.